Foundation/URLCache.swift

```diff
@@ -248,11 +289,101 @@ open class URLCache : NSObject {
         usage of the on-disk cache. 
         @result the current usage of the on-disk cache of the receiver.
     */
-    open var currentDiskUsage: Int { NSUnimplemented() }
+    open var currentDiskUsage: Int {
+        return self.syncQ.sync { self.persistence?.cacheSizeInDisk ?? 0 }
+    }
+
 }
 
 extension URLCache {
     public func storeCachedResponse(_ cachedResponse: CachedURLResponse, for dataTask: URLSessionDataTask) { NSUnimplemented() }
     public func getCachedResponse(for dataTask: URLSessionDataTask, completionHandler: (CachedURLResponse?) -> Void) { NSUnimplemented() }
     public func removeCachedResponse(for dataTask: URLSessionDataTask) { NSUnimplemented() }
 }
+
+fileprivate struct _CachePersistence {
+
+    let path: String
+
+    // FIXME: Create a stored property
+    // Update this value as the cache added and evicted
+    var cacheSizeInDisk: Int {
+        do {
+            let subFiles = try FileManager.default.subpathsOfDirectory(atPath: path)
+            var total: Int = 0
+            for fileName in subFiles {
+                let attributes = try FileManager.default.attributesOfItem(atPath: path.appending(fileName))
+                total += (attributes[.size] as? Int) ?? 0
+            }
+            return total
+        } catch {
+            return 0
+        }
+    }
+
+    func setupCacheDirectory() {
+        do {
+            try FileManager.default.createDirectory(atPath: path, withIntermediateDirectories: true)
+        } catch {
+            fatalError("Unable to create directories for URLCache: \(error.localizedDescription)")
+        }
+    }
+
+    func saveCachedResponse(_ response: CachedURLResponse, for request: URLRequest) {
+        let archivedData = NSKeyedArchiver.archivedData(withRootObject: response)
+        do {
+            if let fileIdentifier = request.cacheFileIdentifier {
+                let cacheFileURL = urlForFileIdentifier(fileIdentifier)
+                try archivedData.write(to: cacheFileURL)
+            }
+        } catch {
+            fatalError("Unable to save cache data: \(error.localizedDescription)")
+        }
+    }
+
+    func cachedResponse(for request: URLRequest) -> CachedURLResponse? {
+        guard let fileIdentifier = request.cacheFileIdentifier else {
+            return nil
+        }
+
+        let url = urlForFileIdentifier(fileIdentifier)
+        guard let data = try? Data(contentsOf: url),
+            let response = NSKeyedUnarchiver.unarchiveObject(with: data) as? CachedURLResponse else {
```

Comment:
Please use secure coding here.

Reply:
To use secure coding, create a `NSKeyedUnarchiver`, set its `requiresSecureCoding` to `true`, and use its method to encode an object for key `NSKeyedArchiveRootObjectKey`.